migration/spec.go

```diff
@@ -0,0 +1,29 @@
+package migration
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// Migrator implements a migration process from one object vesion to another.
+type Migrator interface {
```

Comment:
The idea here would be to have migration "services" in the operators implementing the migrator interface. Each migrator is configured with the appropriate k8s rest client and cache informer to be able to list custom objects and create/delete new/old objects.